### PR TITLE
Bug/test eagle dp v0

### PR DESCRIPTION
**Pull request by Monishver11**
#### Flaky Tests in EAGLE DP

Issue: [#31913](https://github.com/vllm-project/vllm/issues/31913)

This PR resolves the flaky EAGLE DP test in the continuous integration (CI) environment. 
For additional context, related issues and fixes can be found here: https://github.com/pytorch/pytorch/issues/170563.

---

Root Cause: Incomplete batch invariance coverage on L4 GPUs under torch.compile. Two gaps were identified and fixed:

1) UnquantizedEmbeddingMethod.apply (lm_head projection)

UnquantizedLinearMethod.apply (used by all model linear layers) checks VLLM_BATCH_INVARIANT and routes to the deterministic Triton persistent kernel. But UnquantizedEmbeddingMethod.apply (used by ParallelLMHead for the final hidden→vocab logits projection) was missing this check, it always used dispatch_unquantized_gemm() (cuBLAS), which produces batch-size-dependent results on L4.
File: vllm/model_executor/layers/vocab_parallel_embedding.py

2) Batch invariance + torch.compile + CUDA graphs on SM<90 

With enforce_eager=False, both torch.compile and CUDA graph capture are active. On SM<90 (e.g., L4), batch invariance does not work reliably under this combination, the test diverges at varying token positions across runs.
I initially suspected this was due to RMSNorm dispatching to forward_native under custom_ops=["none"], with Inductor codegenning a non-batch-invariant reduction kernel. But an inline repro on L4 in CI showed that torch.compile(rms_norm_native) is actually batch invariant for batch=1 vs batch=4 of the same input row (bitwise_equal: True, max_abs_diff: 0.0), so RMSNorm itself isn't the fault.
This turned out to be a known issue. PR #30018 added enforce_eager=IS_DEVICE_CAPABILITY_BELOW_90 to all 5 batch invariance tests, with the comment in tests/v1/determinism/utils.py:

Note: For devices with SM < 90, batch invariance does not support CUDA Graphs.

The EAGLE DP test was the only batch invariance test that didn't follow this pattern, it had enforce_eager=False unconditionally.
The fix matches the existing pattern: enforce_eager=IS_DEVICE_CAPABILITY_BELOW_90. On L4 (SM89) this disables both torch.compile and CUDA graphs; on H200 (SM90) it stays False since Hopper+ doesn't hit the limitation.
I also tested cudagraph_mode=NONE with enforce_eager=False to isolate which optimization is the actual culprit, the test still failed at token 80. So both torch.compile and CUDA graphs contribute on SM<90, and disabling just one isn't enough. The comment in utils.py may be incomplete, torch.compile is also implicated, not just CUDA graphs. The broader tracking is in [#39096](updated).

Plus, adding the EAGLE DP tests to the Distributed Tests (2 GPUs)(H100) group. The PR includes the relevant changes for this as well.

### CODE REVIEW

File: vllm/v1/core/sched/async_scheduler.py [hunk 1]
```diff
@@ -33,6 +34,23 @@ def _update_after_schedule(self, scheduler_output: SchedulerOutput) -> None:
...
+                    "num_output=%d",
+                    req_id,
+                    request.num_computed_tokens,
+                    request.num_tokens,
+                    request.num_output_placeholders,
+                    request.spec_token_ids[:5],
+                    request.is_prefill_chunk,
+                    scheduler_output.num_scheduled_tokens[req_id],
+                    request.num_output_tokens,
+                )
```
[the remark this answers is not in the document]
Comment (Monishver11) [reply vllm/v1/core/sched/async_scheduler.py]: Removed the loggers.

File: vllm/v1/worker/gpu/model_runner.py [hunk 2]
```diff
@@ -834,6 +846,18 @@ def sample(
...
+            topk_vals, topk_ids = torch.topk(first_logits.float(), 5)
+            logger.info(
+                "[spec_decode_debug][logits] reqs=%d draft=%d "
+                "argmax=%d top5_ids=%s top5_vals=%s",
+                input_batch.num_reqs,
+                input_batch.num_draft_tokens,
+                first_logits.argmax().item(),
+                topk_ids.tolist(),
+                ["{:.6f}".format(v) for v in topk_vals.tolist()],
+            )
```
[the remark this answers is not in the document]
Comment (Monishver11) [reply vllm/v1/worker/gpu/model_runner.py]: Removed the loggers.

File: vllm/v1/core/sched/scheduler.py [hunk 3]
```diff
@@ -234,13 +234,6 @@ def __init__(
             hash_block_size=self.block_size,
             metrics_collector=self.kv_metrics_collector,
         )
-        # Bind GPU block pool to the KV connector. This must happen after
```
Comment (ProExpertProg) [inline vllm/v1/core/sched/scheduler.py:237]: Is this change intentional?

**Review by ProExpertProg — COMMENTED** [review]
Looks good! I think there's a leftover change that doesn't belong though

**Review by ProExpertProg — APPROVED** [review]

**Review by yewentao256 — APPROVED** [review]
LGTM, thanks for the work!

**Review by NickLucche — COMMENTED** [review]
@Monishver11  deserialization issue with tests, so unrelated to this PR. Let's re-run CI.

**Comment by Monishver11** [comment]
As discussed with @ProExpertProg, we will keep the tests under L4 GPUs and also add them to the Distributed Tests (2 GPUs) (H100) group.

File: .buildkite/test_areas/distributed.yaml [hunk 4]
```diff
@@ -191,11 +191,22 @@ steps:
   optional: true
   working_dir: "/vllm-workspace/"
   num_devices: 2
+  source_file_dependencies:
```
Comment (ProExpertProg) [inline .buildkite/test_areas/distributed.yaml]: Source file deps not need for optional job

File: .buildkite/test_areas/distributed.yaml [hunk 5]
```diff
@@ -191,11 +191,22 @@ steps:
...
+  - vllm/v1/spec_decode/
+  - vllm/model_executor/layers/batch_invariant.py
+  - vllm/model_executor/layers/vocab_parallel_embedding.py
+  - tests/v1/distributed/test_eagle_dp.py
   commands:
     - pytest -v -s tests/distributed/test_context_parallel.py
     - VLLM_ALLOW_INSECURE_SERIALIZATION=1 python3 examples/rl/rlhf_async_new_apis.py
     - VLLM_USE_DEEP_GEMM=1 VLLM_LOGGING_LEVEL=DEBUG python3 examples/offline_inference/data_parallel.py --model=Qwen/Qwen1.5-MoE-A2.7B -tp=1 -dp=2 --max-model-len=2048 --all2all-backend=deepep_high_throughput
     - pytest -v -s tests/v1/distributed/test_dbo.py
+    # https://github.com/NVIDIA/nccl/issues/1838
```
Comment (ProExpertProg) [inline .buildkite/test_areas/distributed.yaml]: Are we sure this nccl issue is relevant to this test & hardware?

File: .buildkite/test_areas/distributed.yaml (hunk 4 above)
Comment (Monishver11) [reply .buildkite/test_areas/distributed.yaml]: removed.

File: .buildkite/test_areas/distributed.yaml (hunk 5 above)
Comment (Monishver11) [reply .buildkite/test_areas/distributed.yaml]: This is required for CUDA versions 12.6, 12.8, and 12.9, along with the exact 56x and 57x driver branches. Can you please confirm that H100 groups do not use these versions?

Comment (Monishver11) [reply .buildkite/test_areas/distributed.yaml]: I've checked this, the H100s in CI use SXM and have drivers 13.0. Therefore, this fix isn't needed. I've removed it.

**Review by ProExpertProg — APPROVED** [review]
